File: Data Structures & Algorithms/valid-binary-search-tree/submission-4.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isValidBST(TreeNode root) {
        return isValid(root, new int[]{Integer.MIN_VALUE, Integer.MAX_VALUE});
    }

    public boolean isValid(TreeNode root, int[] range) {
        if (root == null) return true;
        if (root.val > range[0] && root.val < range[1]) {
            return isValid(root.left, new int[]{range[0], root.val}) &&
                    isValid(root.right, new int[]{root.val, range[1]});
        } else {
            return false;
        }
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []

        if not root:
            return res

        q = deque()
        q.append(root)

        while q:
            curLevel = []

            size = len(q)

            for _ in range(size):
                cur = q.popleft()
                curLevel.append(cur.val)
                if cur.left:
                    q.append(cur.left)
                if cur.right:
                    q.append(cur.right)
            res.append(curLevel)

        return res
